lib/widgets/actions.dart

```diff
@@ -18,49 +18,59 @@ import '../model/narrow.dart';
 import 'dialog.dart';
 import 'store.dart';
 
-Future<void> markNarrowAsRead(BuildContext context, Narrow narrow) async {
-  final store = PerAccountStoreWidget.of(context);
-  final connection = store.connection;
-  final zulipLocalizations = ZulipLocalizations.of(context);
-  final useLegacy = connection.zulipFeatureLevel! < 155; // TODO(server-6)
-  if (useLegacy) {
-    try {
-      await _legacyMarkNarrowAsRead(context, narrow);
-      return;
-    } catch (e) {
-      if (!context.mounted) return;
-      showErrorDialog(context: context,
-        title: zulipLocalizations.errorMarkAsReadFailedTitle,
-        message: e.toString()); // TODO(#741): extract user-facing message better
-      return;
+/// High-level operations that interact with the Zulip API and handle UI feedback.
```

Comment:
"handle feedback" sounds like something that *receives* feedback and does something about it. These do the opposite, showing feedback to the user.